awaitility/src/main/java/com/jayway/awaitility/core/AssertionCondition.java

```diff
@@ -118,3 +137,25 @@ public Void await() {
         return null;
     }
 }
+
+class ThrowingRunnableAdapter implements ThrowingRunnable {
+
+    private final Runnable supplier;
+
+    public ThrowingRunnableAdapter(Runnable supplier) {
+        if (supplier == null) {
+            throw new IllegalArgumentException("You must specify a supplier (was null).");
+        }
+
+        this.supplier = supplier;
+    }
+
+    @Override
```

Comment:
Btw, Awaitility is compiled in Java 5 compatibility code. `@Override` for interfaces was introduced in Java 6.